intermediate_source/named_tensor_tutorial.py

```diff
@@ -0,0 +1,518 @@
+# -*- coding: utf-8 -*-
+"""
+Introduction to Named Tensors in PyTorch
+****************************************
+**Author**: `Richard Zou <https://github.com/zou3519>`_
+
+Named Tensors aim to make tensors easier to use by allowing users to associate explicit names
+with tensor dimensions. In most cases, operations that take dimension parameters will accept
+dimension names, avoiding the need to track dimensions by position. In addition, named tensors
+use names to automatically check that APIs are being used correctly at runtime, providing extra
+safety. Names can also be used to rearrange dimensions, for example, to support
+"broadcasting by name" rather than "broadcasting by position".
+
+this tutorial is intended as a guide to the functionality that will
+be included with the 1.3 launch. By the end of it, you will be able to:
+
+- Initiate a ``Tensor`` with named dimensions, as well as removing or renmaing those dimensions
+- Understand the basics of how dimension names are propagated through operations
+- See how naming dimensions enables clearer code in two key areas:
+    - Broadcasting operations
+    - Flattening and unflattening dimensions
+
+Finally, we'll put this into practice by writing a multi-headed attention module

```

Comment:
Let's and call it "multi-head attention" rather than "multi-headed attention". I think more people call it the former (googling "multi-headed attention" leads to search results for "multi-head attention")